singularity-eos/eos/eos.hpp

```diff
@@ -88,6 +122,144 @@ class IdealGas {
                               Real &bmod, Real &dpde, Real &dvdt,
                               Real *lambda = nullptr) const;
 
+  /*
+  Vector versions of the member functions are only available for CPU
+  
+  RealIndexer must have an operator[](int) that returns a Real. e.g., Real*
+  ConstRealIndexer is as RealIndexer, but assumed const type.
+  LambdaIndexer must have an operator[](int) that returns a Real*. e.g., Real**
+  */
+  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
+  inline
+  void TemperatureFromDensityInternalEnergyVector(ConstRealIndexer &&rhos,
```

Comment:
No reason to name these with the `Vector` postfix. Just let multiple dispatch handle it. Because the arguments are different, the functions will be different.